pontoon/base/models.py

```diff
@@ -25,12 +25,72 @@
 log = logging.getLogger('pontoon')
 
 
+class UserTranslationsManager(models.Manager):
+    """
+    Provides various method to interact with larger sets of translations and their stats for user.
+    """
+
+    def _changed_translations_count(self, args):
+        """
+        Helper method, returns expression object which allows us to annotate querysets
+        with counts of translations.
+        """
+        translation_query = (
+            ~Q(translation__string=F('translation__entity__string'))
+            & ~Q(translation__string=F('translation__entity__string_plural'))
+            & Q(translation__user__isnull=False)
```

Comment:
yeah, it was really weird. However, i'll start by writing more tests and
then we'll if they help to pinpoint more possible suspect.

On Fri, Sep 11, 2015 at 5:12 PM, Michael Kelly <notifications@github.com>
wrote:

> In pontoon/base/models.py
> <https://github.com/mozilla/pontoon/pull/161#discussion_r39281036>:
>
> > @@ -25,12 +25,72 @@
> >  log = logging.getLogger('pontoon')
> >
> >
> > +class UserTranslationsManager(models.Manager):
> > +    """
> > +    Provides various method to interact with larger sets of translations and their stats for user.
> > +    """
> > +
> > +    def _changed_translations_count(self, args):
> > +        """
> > +        Helper method, returns expression object which allows us to annotate querysets
> > +        with counts of translations.
> > +        """
> > +        translation_query = (
> > +            ~Q(translation__string=F('translation__entity__string'))
> > +            & ~Q(translation__string=F('translation__entity__string_plural'))
> > +            & Q(translation__user__isnull=False)
>
> wat
>
> That's really odd. I guess leave it in for now, and we can take a closer
> look later. If this works as expected and the tests pass, it probably
> doesn't have any side effects. So either I'm wrong (very very possible) or
> something else is causing the issue and this just happens to fix it. Either
> possibility can be resolved later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/pontoon/pull/161/files#r39281036>.
>



-- 
jarek@reijutsu:~$ fortune
You have Egyptian flu: you're going to be a mummy.
jarek@reijutsu:~$ fortune
You now have Asian Flu.